### PR TITLE
 feat(#187): create order details page with nested dynamic paths

### PR DESCRIPTION
**Pull request by letelete**
With this commit, each order has its own path in the app router `/my-account/order-details/<order-id>`.

## Description

I've transformed reactive order details/order history in the `OrderHistory` file into two separate files that support dynamic routing: `OrderHistory` and `OrderDetails`.

I've added `extendRoutes` function to the `nuxt.config.js` which supports /order-details/<id>. This change required refactoring `MyAccount` page.

I've added order-details middleware to disallow user from opening `/my-account/order-details` or `/my-account/order-details/<INCORRECT_ORDER_ID>` - in this case, user is being redirected to the `/my-account/order-history`

## Related Issue

#187 

## Motivation and Context

Previously, the navigation was unintuitive. The reactive state wouldn't support proper behaviors for navigating back while browsing orders history and order details.

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/18665370/161102019-e9ddddb5-1957-4dc5-ba91-3327e44f5d02.mov

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by letelete** [comment]
Not sure why Commitlint fails.

File: packages/theme/middleware/order-details.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export default async ({ app, $vsf }) => {
+  const callFallbackRedirect = () => app.context.redirect('/my-account/order-history');
+
+  const { pageName, id: orderId } = app.context.route.params;
+  if (pageName !== 'order-details') return;
+  if (!orderId) return callFallbackRedirect();
+
+  const orders = await $vsf.$spree.api.getOrders();
+  if (orders.some(order => orderGetters.getId(order)?.toLowerCase() === orderId)) return;
```
Comment (rafalcymerys) [inline packages/theme/middleware/order-details.js:11]: This seems a bit too heavy - no point in pulling all orders if we want to check a single order. 
There's an endpoint for pulling a single order by number: https://api.spreecommerce.org/docs/api-v2/b3A6MTgwNTI4NjA-retrieve-an-order

We could add an endpoint in our api-client package to support it and then use it here.

File: packages/theme/nuxt.config.js [hunk 2]
```diff
@@ -132,9 +132,8 @@ export default {
             currencyDefault: 'EUR'
           }
         }
-      },
-    },
-    detectBrowserLanguage: false
```
Comment (rafalcymerys) [inline packages/theme/nuxt.config.js:137]: Why did you remove detectBrowserLanguage?

File: packages/theme/middleware/order-details.js (hunk 1 above)
Comment (rafalcymerys) [reply packages/theme/middleware/order-details.js:11]: Actually, should we do a redirect if an order is not found? Maybe we should just let the customer enter the page and show a message "This order could not be found"?

File: packages/theme/pages/MyAccount/OrderHistory.vue [hunk 3]
```diff
@@ -82,27 +41,29 @@
...
+    SfButton
   },
-  setup() {
+  setup(_, context) {
+    const router = useRouter();
     const { orders, search } = useUserOrder();
-    const currentOrder = ref(null);
+
+    const displayOrderDetails = (order) => {
+      const path = `/my-account/order-details/${orderGetters.getId(order).toLowerCase()}`;
```
Comment (rafalcymerys) [inline packages/theme/pages/MyAccount/OrderHistory.vue:63]: Maybe we should use order's number instead of id in the path? It's customer facing

File: packages/theme/nuxt.config.js (hunk 2 above)
Comment (rafalcymerys) [reply packages/theme/nuxt.config.js:137]: Ok, I see it is already defined above. Let's keep this change

File: packages/theme/pages/MyAccount/OrderHistory.vue (hunk 3 above)
Comment (rafalcymerys) [reply packages/theme/pages/MyAccount/OrderHistory.vue:63]: Nevermind, I see it's the order number

File: packages/theme/middleware/order-details.js (hunk 1 above)
Comment (letelete) [reply packages/theme/middleware/order-details.js]: > This seems a bit too heavy - no point in pulling all orders if we want to check a single order. There's an endpoint for pulling a single order by number: https://api.spreecommerce.org/docs/api-v2/b3A6MTgwNTI4NjA-retrieve-an-order
> 
> We could add an endpoint in our api-client package to support it and then use it here.

I've added the getOrder API method and updated useUserOrder accordingly.

Comment (letelete) [reply packages/theme/middleware/order-details.js]: > Actually, should we do a redirect if an order is not found? Maybe we should just let the customer enter the page and show a message "This order could not be found"?

I've changed the redirection logic into raising the 404 error (which displays the error.vue layout) in the middleware.

**Review by rafalcymerys — APPROVED** [review]
